models/nc-news.models.js

```diff
@@ -12,26 +12,38 @@ exports.fetchTopics = () => {
     });
 };
 
-exports.fetchArticles = (articleId) => {
-  return db
-    .query(
-      `
-    SELECT articles.article_id, articles.title, users.name AS author, articles.body, articles.topic, articles.created_at, articles.votes, COUNT(comments.article_id)::int AS comment_count
-    FROM articles
-    JOIN users ON articles.author = users.username
-    JOIN comments ON comments.article_id = articles.article_id
-    WHERE articles.article_id = $1
-    GROUP BY articles.article_id, users.name;
-    `,
-      [articleId]
-    )
-    .then(({ rows }) => {
-      if (rows.length === 0) {
-        return Promise.reject({ status: 404, msg: "not found" });
-      } else {
-        return rows[0];
-      }
-    });
+exports.fetchArticles = (articleId, topicQuery) => {
+  let queryStr = `
+  SELECT articles.article_id, articles.title, users.name AS author, articles.body, articles.topic, articles.created_at, articles.votes, COUNT(comments.article_id)::INT AS comment_count
+  FROM articles
+  JOIN users ON articles.author = users.username
+  LEFT JOIN comments ON comments.article_id = articles.article_id`;
```

Comment:
This join is correct. Ignore my previous one. I still dont think you need the author join though